samcli/cli/types.py

```diff
@@ -25,11 +36,8 @@ class CfnParameterOverridesType(click.ParamType):
     # If Both ParameterKey pattern and KeyPairName=MyKey should not be present
     # while adding parameter overrides, if they are, it
     # can result in unpredicatable behavior.
-    KEY_REGEX = '([A-Za-z0-9\\"]+)'
-    VALUE_REGEX = '(\\"(?:\\\\.|[^\\"\\\\]+)*\\"|(?:\\\\.|[^ \\"\\\\]+)+))'
-
-    _pattern_1 = r"(?:ParameterKey={key},ParameterValue={value}".format(key=KEY_REGEX, value=VALUE_REGEX)
-    _pattern_2 = r"(?:(?: ){key}={value}".format(key=KEY_REGEX, value=VALUE_REGEX)
+    _pattern_1 = r"(?:ParameterKey={key},ParameterValue={value})".format(key=KEY_REGEX, value=VALUE_REGEX_SPACE_DELIM)
+    _pattern_2 = r"(?:(?: ){key}={value})".format(key=KEY_REGEX, value=VALUE_REGEX_SPACE_DELIM)
```

Comment:
Works well, Gotchas with quotes.

Works
` dd=gfbfg asdf=dadsf asdasd=qwe` Groups (dd,gfbfg) (asdf,dadsf) (asdasd, qwe)
`" "dd"='gfbfg'"` Groups ("dd",'gfbfg')

Doesn't work

`' "dd"="gfbfg" "asdf"="dadsf" "asdasd"="qwed"'` No Groups

Reply:
I am not worried about the key not supporting quotes. that's out of scope. Also, value using single quotes is also out-of-scope.